pmd-core/src/main/java/net/sourceforge/pmd/autofixes/operations/TextOperation.java

```diff
@@ -95,70 +94,26 @@ private boolean containsChildRegion(final TextOperation childToAdd) {
     }
 
     private int getIndexToAddChild(final TextOperation childToAdd) {
-        final int index;
+        int potentialIndex = Collections.binarySearch(children, childToAdd);
 
-        if (children.isEmpty()) {
-            index = 0;
-        } else if (startsAfterLastChild(childToAdd)) {
-            index = children.size();
-        } else {
-            int potentialIndex = Collections.binarySearch(children, childToAdd);
-
-            if (potentialIndex < 0) {
-                index = -(potentialIndex + 1);
-            } else {
-                final int childrenLastIndex = children.size() - 1;
+        if (potentialIndex < 0) { // It is not an insert operation
+            return -(potentialIndex + 1);
```

Comment:
`return ~potentialIndex`?

Reply:
According to the [Collections.binarySearch method's documentation](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html), the return value is defined as follows: 
> The index of the search key, if it is contained in the list; otherwise, (-(insertion point) - 1). The insertion point is defined as the point at which the key would be inserted into the list: [...]

Reply:
I know, that's why I say it... for signed variables under two's complement, you just invert every bit...

Reply:
Oh my god, I just realised you were using the bitwise operator, I couldn't make any sense of inverting every bit with just a substraction op. Brillant.